src/v/model/record_batch_reader.h

```diff
@@ -330,9 +336,32 @@ record_batch_reader make_generating_record_batch_reader(
 ss::future<record_batch_reader::data_t> consume_reader_to_memory(
   record_batch_reader, timeout_clock::time_point timeout);
 
-ss::future<fragmented_vector<model::record_batch>>
+template<size_t fragment_size_bytes = 8192>
+ss::future<fragmented_vector<model::record_batch, fragment_size_bytes>>
```

Comment:
should we keep this in the implementation file too? we probably dont need to support something other than the default 8k right?

Reply:
good point, i will change this to do the same trick as with `make_fragmented...` methods